packages/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/configurators/DevLauncherExpoActivityConfigurator.kt

```diff
@@ -144,4 +147,61 @@ class DevLauncherExpoActivityConfigurator(
     activity
       .window.statusBarColor = color
   }
+
+  fun applyNavigationBarConfiguration(activity: ReactActivity) {
+    val navBarOptions = manifest.getAndroidNavigationBarOptions() ?: return
+
+    // Set background color of navigation bar
+    val navBarColor = navBarOptions.optString("backgroundColor")
+    if (isValidColor(navBarColor)) {
+      try {
+        activity.window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION)
+        activity.window.navigationBarColor = Color.parseColor(navBarColor)
+      } catch (e: Throwable) {
+        Log.e(TAG, "Failed to configure androidNavigationBar.backgroundColor", e)
+      }
+    }
+
+    // Set icon color of navigation bar
+    val navBarAppearance = navBarOptions.optString("barStyle")
+    if (navBarAppearance != "" && Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      try {
+        activity.window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION)
+        activity.window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
+        if (navBarAppearance == DevLauncherNavigationBarStyle.DARK) {
+          val decorView = activity.window.decorView
+          var flags = decorView.systemUiVisibility
+          flags = flags or View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
+          decorView.systemUiVisibility = flags
+        }
+      } catch (e: Throwable) {
+        Log.e(TAG, "Failed to configure androidNavigationBar.barStyle", e)
+      }
+    }
+
+    // Set visibility of navigation bar
+    val navBarVisible = navBarOptions.optString("visible")
+    if (navBarVisible != "") {
+      // Hide both the navigation bar and the status bar. The Android docs recommend, "you should
+      // design your app to hide the status bar whenever you hide the navigation bar."
+      val decorView = activity.window.decorView
+      var flags = decorView.systemUiVisibility
+      when (navBarVisible) {
+        DevLauncherNavigationBarVisibility.LEANBACK ->
+          flags =
+            flags or (View.SYSTEM_UI_FLAG_HIDE_NAVIGATION or View.SYSTEM_UI_FLAG_FULLSCREEN)
+        DevLauncherNavigationBarVisibility.IMMERSIVE ->
+          flags =
+            flags or (View.SYSTEM_UI_FLAG_HIDE_NAVIGATION or View.SYSTEM_UI_FLAG_FULLSCREEN or View.SYSTEM_UI_FLAG_IMMERSIVE)
+        DevLauncherNavigationBarVisibility.STICKY_IMMERSIVE ->
+          flags =
+            flags or (View.SYSTEM_UI_FLAG_HIDE_NAVIGATION or View.SYSTEM_UI_FLAG_FULLSCREEN or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY)
+      }
```

Comment:
```suggestion
      val flags = decorView.systemUiVisibility or when (navBarVisible) {
        DevLauncherNavigationBarVisibility.LEANBACK -> (View.SYSTEM_UI_FLAG_HIDE_NAVIGATION or View.SYSTEM_UI_FLAG_FULLSCREEN)
        DevLauncherNavigationBarVisibility.IMMERSIVE -> (View.SYSTEM_UI_FLAG_HIDE_NAVIGATION or View.SYSTEM_UI_FLAG_FULLSCREEN or View.SYSTEM_UI_FLAG_IMMERSIVE)
        DevLauncherNavigationBarVisibility.STICKY_IMMERSIVE -> (View.SYSTEM_UI_FLAG_HIDE_NAVIGATION or View.SYSTEM_UI_FLAG_FULLSCREEN or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY)
      }
```
IMO, looks better but up to you ;)